### DISCUSSION TITLE
Looser esbuild semver pattern

### DISCUSSION THREAD

**Discussion by tmcw** [resend/react-email#921] [Ideas]
### Goals

1. I would like to avoid having multiple copies of esbuild installed in my project

### Background

In the bad old days, we'd end up with multiple webpacks. We'd like to avoid that in the future.

### Proposal

react-email pins all of its npm dependencies:

https://github.com/resendlabs/react-email/blob/main/packages/react-email/package.json#L37

which might be good for reliability, but is basically bad for the `node_modules` directory, especially in the case of esbuild, in which it's pinning a particular out-of-date version. It would be great to use looser, and newer, semver specifiers to allow `npm` to do its work of deduplicating big dependencies like esbuild.

**Comment by shellscape** [comment]
Agreed. Was one of the motivating factors for an effort I started to write a replacement for the CLI. About a week or so out from making that public. If you'd like to see progress, you can check that out on my fork here: https://github.com/shellscape/jsx-email/tree/feat/cli/initial
